### PR TITLE
[WIP] Fix race conditions in OSM.org data sync

### PR DESCRIPTION
**Pull request by noffle**

### CODE REVIEW

**Comment by noffle** [comment]
Fixes #168.

Do not merge until https://github.com/substack/hyperlog-kdb-index/pull/2 is merged and published.

File: app/actions/index.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ const types = {
   FINISHED_SYNCING_SURVEY_DATA: "FINISHED_SYNCING_SURVEY_DATA",
   SET_AREA_OF_INTEREST: "SET_AREA_OF_INTEREST",
   UPDATE_OBSERVATION: "UPDATE_OBSERVATION"
```
Comment (mojodna) [inline app/actions/index.js]: missing comma

File: app/actions/index.js [hunk 2]
```diff
@@ -129,49 +130,65 @@ const checkOsmMeta = (url, getState, cb) => {
...
+        },
+        err => {
+          if (err) {
+            return dispatch({
+              type: types.SYNCING_SURVEY_DATA_FAILED,
+              id
+            });
+          }
+
+          console.log("osm.replicate", err);
```
Comment (mojodna) [inline app/actions/index.js]: `err` is `null`.

File: app/actions/index.js [hunk 3]
```diff
@@ -129,49 +130,65 @@ const checkOsmMeta = (url, getState, cb) => {
...
-
-      dispatch({
-        type: types.SET_AREA_OF_INTEREST,
-        areaOfInterest
-      });
-    });
+export const destroyAllData = () => dispatch => {
+  console.log("destroy all data");
+  dispatch({ type: types.CLEAR_LOCAL_SURVEYS });
+  dispatch({ type: types.CLEAR_REMOTE_SURVEYS });
```
Comment (mojodna) [inline app/actions/index.js]: Another way to do [some] of this is to call `.purge()` on the result of `persistStore()` in https://github.com/osmlab/field-data-collection/blob/master/app/config/store.js#L20-L23 That's how I reset survey definition state.

File: app/actions/index.js (hunk 1 above)
Comment (noffle) [inline app/actions/index.js]: Thanks.

File: app/actions/index.js (hunk 2 above)
Comment (noffle) [inline app/actions/index.js]: Good catch.

File: app/actions/index.js (hunk 3 above)
Comment (noffle) [inline app/actions/index.js]: I'll leave this revision in @sethvincent's hands.

**Comment by sethvincent** [comment]
@noffle Let's use the fork temporarily and make an issue to update to the published version when it happens.

**Comment by sethvincent** [comment]
Nevermind, I see it was just published!

**Comment by noffle** [comment]
@sethvincent Do we need to update `yarn.lock` to point to the latest release?

**Pull request merged**